wallet/tickets.go

```diff
@@ -25,6 +26,27 @@ func sliceContainsHash(s []chainhash.Hash, h chainhash.Hash) bool {
 	return false
 }
 
+// GenerateVoteTx returns a hash
+func (w *Wallet) GenerateVoteTx(blockHash *chainhash.Hash, height int64, sstxHash *chainhash.Hash, voteBits stake.VoteBits) (*wire.MsgTx, error) {
+	var msgtx *wire.MsgTx
+	err := walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
+		stakemgrNs := tx.ReadWriteBucket(wstakemgrNamespaceKey)
+		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
+
+		shit, err := w.StakeMgr.GenerateVoteTx(stakemgrNs, addrmgrNs, blockHash, height, sstxHash, voteBits)
+		if err != nil {
+			return err
+		}
+		msgtx = shit
```

Comment:
Yeah, this isn't a bug.  I added the View/Update apis and would have made it a lot nicer when trying to return data out of the functions, but this is sadly not possible to do in a type safe way without generics.

For example, if we could specify a generic type parameter T, the update func signature could be:

`func Update(db DB, func(tx ReadWriteTx) T) T`

and then it would not be required to close over value just to use it as what is effectively the update's return value.